### PR TITLE
Add "Getting Started" tutorial

### PR DESCRIPTION
**Pull request by SimonHeybrock**
This is based on (but modified) what we did in the Scipp courses.

### CODE REVIEW

**Comment by jl-wynen** [comment]
Is this supposed to be used as an alternative to the other 'Getting Started' docs? Or as a replacement for quick start? It seems to have a lot of overlap.

**Comment by SimonHeybrock** [comment]
> Is this supposed to be used as an alternative to the other 'Getting Started' docs? Or as a replacement for quick start? It seems to have a lot of overlap.

Not sure, but I thought we need a tutorial that starts at zero.

File: docs/tutorials/getting-started.ipynb [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b6bc810c-ec39-4c9a-901f-b66d88a40782",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "length = sc.scalar(50.0, variance=0.01, unit='mm')\n",
```
Comment (nvaytet) [inline docs/tutorials/getting-started.ipynb:193]: I think it's a little confusing to people who read this for the first time that we were using `1.2 * sc.Unit('m')` above, and `sc.scalar` here. I suggest to only use `sc.scalar()`, and maybe mention as a side note, e.g. after the first use of `scalar`, that you can also use the `*` syntax as an alternative.

I also think that many python beginners will struggle to fully grasp what `sc.Unit('m')` is actually doing/representing.

File: docs/tutorials/getting-started.ipynb [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0fddf047-538e-42b0-9965-59e2722ad067",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "var['time', 2] + var['location', 2]"
```
Comment (nvaytet) [inline docs/tutorials/getting-started.ipynb:331]: I think automatic broadcasting is a bit too much to have in such a low-level tutorial. Maybe just extracting a single column is enough?

File: docs/tutorials/getting-started.ipynb [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+   "metadata": {
+    "tags": [
+     "raises-exception"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "other = da.copy()  # Copy and modify, for illustration purposes\n",
+    "other.coords['time'][-1] += 0.1 * sc.Unit('s')\n",
+    "da + other  # will raise an exception"
```
Comment (nvaytet) [inline docs/tutorials/getting-started.ipynb:406]: This will make the CI fail?

File: docs/tutorials/getting-started.ipynb [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+    "stop = 1.4 * sc.Unit('s')\n",
+    "da['time', start:stop]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2802cfc9-4599-4602-8c9b-20018b0ed807",
+   "metadata": {},
+   "source": [
+    "In addition to coordinates, data arrays also provide a dictionary of attributes and a dictionary of masks."
```
Comment (nvaytet) [inline docs/tutorials/getting-started.ipynb:444]: I think we should either remove this sentence, or add explicit examples for illustration.

**Review by nvaytet — COMMENTED** [review]
Looks good overall, some small comments

**Comment by SimonHeybrock** [comment]
> So are people expected to read through the getting started section and then the tutorial? Or is the tutorial meant to be used in a workshop or similar?

You mean the "Quick Start" page? I don't think this is really meant for "beginners". But maybe we should remove it if you think it is not useful?

File: docs/tutorials/getting-started.ipynb (hunk 2 above)
Comment (SimonHeybrock) [reply docs/tutorials/getting-started.ipynb:331]: I wanted to illustrate how dimension labels avoid pitfalls of NumPy. Do you have an alternative suggestion?

File: docs/tutorials/getting-started.ipynb (hunk 1 above)
Comment (nvaytet) [reply docs/tutorials/getting-started.ipynb:193]: hmm maybe `quantity` is better?

**Comment by jl-wynen** [comment]
> You mean the "Quick Start" page? I don't think this is really meant for "beginners". But maybe we should remove it if you think it is not useful?

There are 2 reason I asked this:
- We should not duplicate information too much, it just means that we have to update more places in the future. (Some duplication is good, though)
- It should be clear for people visiting the site for the first time, where to start. And we need to know where to send people when we tell them to try scipp. We can have a page explaining this or maybe something at the end of the getting started section to show where to go from there.

**Comment by SimonHeybrock** [comment]
I suggest to change the focus of this tutorial to only using scipp as a "calculator", with 0-D and 1/N-D variables. I will try to do so on a concrete example (a bit like in the table-to-binned tutorial).
The data-array things given here are documented elsewhere, or should move into a (to be written) dense-data tutorial.

**Comment by SimonHeybrock** [comment]
Closing in favor of new PR.

**Pull request closed without merging**